src/test/erasure-code/ceph_erasure_code_benchmark.cc

```diff
@@ -162,7 +171,7 @@ int ErasureCodeBench::encode()
 
   if (erasure_code->get_data_chunk_count() != (unsigned int)k ||
       (erasure_code->get_chunk_count() - erasure_code->get_data_chunk_count()
-       != (unsigned int)m)) {
+       - (l > 0 ? ((k+m)/l) : l) != (unsigned int)m)) {
```

Comment:
in the simplest case, take `k=2,m=1,l=3` as an example, where data and coding chunks will be grouped into a single group of size 3. using your algorithm:
```
3 - 2 - (3 + 1) / 3 = 1 - 1 = 0
```
but m is 1 here. actually, i don't think we can deduce m using the locality (`l`). instead, i'd suggest revert c5bafae6fd10282f408695c7763b1277469123f6

Reply:
@tchaikov ,
In case of LRC get_chunk_count is calculated in different way I think. I ran with gdb with k=2,m=1,l=3
(gdb) p k                                                                                                                                                                   |
$1 = 2
(gdb) p m                                                                                                                                                                   |
$2 = 1
(gdb) p l
$3 = 3
(gdb) p erasure_code->get_chunk_count()                                                                                                                                     
$4 = 4
(gdb) p erasure_code->get_data_chunk_count()
$5 = 2

So its working for me. 

Am I missing something ??

Thanks,
Shylesh

Reply:
@shylesh what does the "mapping" look like in your case?

Reply:
(gdb) p mapping
$11 = "DD__"
(gdb) p layers
$12 = "[  [ \"DDc_\", \"\" ], [ \"DDDc\", \"\" ],"

Reply:
@tchaikov ,

Did you get a chance to look at this. Let me know if I need to do any modifications.

Thanks,
Shylesh

Reply:
no, i have not. will take a look next Monday.

Reply:
@shylesh i'd r you try to this one: http://docs.ceph.com/docs/master/rados/operations/erasure-code-lrc/#minimal-testing . it is the *simplest* case of LRC, in which LRC degrades to a plain erasure, where chunk count is 3, and data chunk count is 2. because the # of `D` in `DD_`  is 2, and ```len(`DD_`)``` is 3.

> Let me know if I need to do any modifications.

sure, they are 

- https://github.com/ceph/ceph/pull/19297/files#r154628195
- instead, i'd suggest revert c5bafae